drivers/tcs37727/tcs37727.c

```diff
@@ -0,0 +1,276 @@
+
+#define ENABLE_DEBUG   (1)
+#include "debug.h"
+
+#define I2C_SPEED      I2C_SPEED_FAST
+
+static int tcs37727_test(tcs37727_t *dev)
+{
+    char id;
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_read_reg(dev->i2c, dev->addr, TCS37727_ID, &id) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (id != TCS37727_ID_VALUE) {
+        return -1;
+    }
+
+    return 0;
+}
+
+int tcs37727_init(tcs37727_t *dev, i2c_t i2c, uint8_t address, int atime_us)
+{
+    /* write device descriptor */
+    dev->i2c = i2c;
+    dev->addr = address;
+    dev->initialized = false;
+
+    i2c_acquire(dev->i2c);
+
+    /* initialize the I2C bus */
+    if (i2c_init_master(i2c, I2C_SPEED) < 0) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+
+    if (tcs37727_test(dev)) {
+        return -2;
+    }
+
+    i2c_acquire(dev->i2c);
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_CONTROL,
+                      TCS37727_CONTROL_AGAIN_4) != 1) {
+        i2c_release(dev->i2c);
+        return -3;
+    }
+    dev->again = 4;
+
+    if (i2c_write_reg(dev->i2c, dev->addr,
+                      TCS37727_ATIME_TO_REG(atime_us), 0) != 1) {
+        i2c_release(dev->i2c);
+        return -3;
+    }
+    dev->atime_us = atime_us;
+
+    dev->initialized = true;
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+int tcs37727_set_rgbc_active(tcs37727_t *dev)
+{
+    uint8_t reg;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    i2c_acquire(dev->i2c);
+    if (i2c_read_regs(dev->i2c, dev->addr, TCS37727_ENABLE, (char *)&reg, 1) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    reg |= (TCS37727_ENABLE_AEN | TCS37727_ENABLE_PON);
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_ENABLE, reg) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+int tcs37727_set_rgbc_standby(tcs37727_t *dev)
+{
+    uint8_t reg;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    i2c_acquire(dev->i2c);
+    if (i2c_read_regs(dev->i2c, dev->addr, TCS37727_ENABLE, (char *)&reg, 1) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    reg &= ~TCS37727_ENABLE_AEN;
+    if (!(reg & TCS37727_ENABLE_PEN)) {
+        reg &= ~TCS37727_ENABLE_PON;
+    }
+
+    if (i2c_write_reg(dev->i2c, dev->addr, TCS37727_ENABLE, reg) != 1) {
+        i2c_release(dev->i2c);
+        return -1;
+    }
+
+    i2c_release(dev->i2c);
+    return 0;
+}
+
+static uint8_t tcs37727_trim_gain(tcs37727_t *dev, int rawc)
+{
+    uint8_t reg_again = 0;
+    int val_again = dev->again;
+
+    if (dev->initialized == false) {
+        return -1;
+    }
+
+    if (rawc < TCS37727_AG_THRESHOLD_LOW) {
+        switch (val_again) {
+            case 1:
+                reg_again = TCS37727_CONTROL_AGAIN_4;
+                val_again = 4;
+                break;
```

Comment:
I don't understand the relation to https://github.com/RIOT-OS/RIOT/pull/3135/files#diff-c13e75366b4489aac1fd522db3c5b4a5R82. There you set ` TCS37727_CONTROL_AGAIN_4` and `dev->again = 4;`. Here you read this value by `val_again = dev->again;` and when the switch case is `4` (like you initialized) you set the parameters `TCS37727_CONTROL_AGAIN_16` and `val_again = 16;`